src/Microsoft.AspNetCore.Authentication.Facebook/FacebookExtensions.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Authentication.Facebook;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class FacebookAuthenticationOptionsExtensions
+    {
+        /// <summary>
+        /// Adds facebook authentication with options bound against the "Facebook" section 
+        /// from the IConfiguration in the service container.
+        /// </summary>
+        /// <param name="services"></param>
+        /// <returns></returns>
+        public static IServiceCollection AddFacebookAuthentication(this IServiceCollection services)
+            => services.AddFacebookAuthentication(FacebookDefaults.AuthenticationScheme, _ => { });
+
+        public static IServiceCollection AddFacebookAuthentication(this IServiceCollection services, Action<FacebookOptions> configureOptions) 
+            => services.AddFacebookAuthentication(FacebookDefaults.AuthenticationScheme, configureOptions);
+
+        public static IServiceCollection AddFacebookAuthentication(this IServiceCollection services, string authenticationScheme, Action<FacebookOptions> configureOptions)
+        {
+            services.AddSingleton<IConfigureOptions<FacebookOptions>, FacebookConfigureOptions>();
```

Comment:
> Well, your delegate runs after the binding so it 'wins'.

Only if the DI container respects the services registration order, right? (`IConfigureOptions.Order` is no longer a thing in 1.0/2.0, so it entirely relies on the DI capabilities)

It would be way clearer if this feature was more explicit (e.g by flowing the `IConfiguration` as a parameter). It would also the developer to select the nodes he/she wants:

```csharp
services.AddFacebookAuthentication(Configuration.GetSection("security:authentication:facebook"));
```